src/commands/project/delete/source.ts

```diff
@@ -355,7 +355,19 @@ export class Source extends SfCommand<DeleteSourceJson> {
             }
           }
           if (component.xml) {
-            promises.push(fsPromises.unlink(component.xml));
```

Comment:
for every custom label in the CS, this calls STL.deleteCustomLabels on every label in the set.

Some of those are going to be targeted multiple times?

Maybe you want to do that filter of labels once at the top of the method rather than inside the per-component loop